### DISCUSSION TITLE
AM -f and install recommendation

### DISCUSSION THREAD

**Discussion by ochi12** [ivan-hc/AM#1034] [Ideas]
It's a bomber that i need to switch to am and appman mode just to show installed apps respectively. 
may it should look like this?

```
$am -f

-  USER $USER INSTALLATION ( 2 package/s are installed) 

-  APPNAME      |  VERSION  |  TYPE            |  SIZE  
-  -------      |  -------  |  ----            |  ----  
◆  arduino-ide  |  2.3.3    |  dynamic-binary  |  182   MiB
◆  inkscape     |  1.4      |  dynamic-binary  |  119   MiB

 TOTAL SIZE: 301 MiB of disk space in use

-------------------------------------------------------------

- SYSTEM INSTALLATION (2 package/s are installed)

-  APPNAME  |  VERSION  |  TYPE            |  SIZE  
-  -------  |  -------  |  ----            |  ----  
◆  nvim     |  0.10.2   |  dynamic-binary  |  12    MiB
◆  am       |  8.4.1    |  bash-script     |  144   KiB

 TOTAL SIZE: 12 MiB of disk space in use
```

```
$ am --user -i ${PACKAGE}
.
.
blabla
.
.

$ am -i ${NEXT_PACAKGE}
-----------------------------------------------------------------------------
 "AM" is running as "AppMan", use `am --system -i` to switch it back to "AM"
-----------------------------------------------------------------------------
.
.
blabla
.
.
```
or maybe by default prompt users to chose system or user installation and other things

**Comment by ivan-hc** [comment]
I have just tested this for `-f` to allow "AM" see installed apps also in AppMan, while "AppMan" or 2AppMan Mode" are not allowed

https://github.com/user-attachments/assets/1311da9f-91fa-42e4-a405-ace2ae3fcbca

the trick is to check if the file "appman-config" exists

https://github.com/user-attachments/assets/c4e9436e-65c3-4a19-ba6b-942cacb3228e

the responsible of this is a new function I added in "database.am"
```
_files_check_appman_from_am() {
	[ "$AMCLI" = am ] && [ -f "$APPMANCONFIG"/appman-config ] && _appman && echo "$DIVIDING_LINE" && APPMAN_ON=1
}
```
that is applied in the option like this
```
	'-f'|'files')
		if [ "$2" = "--less" ]; then
			_files_show_only_number
			_files_check_appman_from_am && _files_show_only_number
		elif [ "$2" = "--byname" ]; then
			_files_sort_by_name
			_files_check_appman_from_am && _files_sort_by_name
			_betatester_message_on
		else
			_files_sort_by_size
			_files_check_appman_from_am && _files_sort_by_size
			_betatester_message_on
		fi
		;;
```
plus some corrections to the messages
```
_files_header() {
	echo ""
	APPSNUMB=$(find "$APPSPATH" -type f -name 'remove' -exec grep -L 'usr/local/lib' {} \; 2>/dev/null | sed -r 's|/[^/]+$||' | sort | uniq | wc -l)
	if [ "$APPSNUMB" = 1 ]; then
		APPSMESSAGE="YOU HAVE INSTALLED $APPSNUMB PROGRAM"
	else
		APPSMESSAGE="YOU HAVE INSTALLED $APPSNUMB PROGRAMS"
	fi
	if grep -q 'usr/local/lib' "$APPSPATH"/*/remove 2> /dev/null; then
		LIBNUMB=$(grep -l "usr/local/lib" "$APPSPATH"/*/remove | wc -l)
		if [ "$LIBNUMB" = 1 ]; then
			LIBSMESSAGE="AND $LIBNUMB LIBRARY"
		else
			LIBSMESSAGE="AND $LIBNUMB LIBRARIES"
		fi
		[ "$APPMAN_ON" = 1 ] && echo "- $APPSMESSAGE $LIBSMESSAGE IN \"APPMAN MODE\", LOCALLY" || echo "- $APPSMESSAGE $LIBSMESSAGE MANAGED BY \"$AMCLIUPPER\""
	else
		[ "$APPMAN_ON" = 1 ] && echo "- $APPSMESSAGE IN \"APPMAN MODE\", LOCALLY" || echo "- $APPSMESSAGE MANAGED BY \"$AMCLIUPPER\""
	fi
	echo ""
}
```
to test all this (and join the beta-testing) run `am --devmode-enable` and `am -s`

@Samueru-sama I'd like you join this effort

**Reply by Samueru-sama** [reply]
I really don't know how the portable mode works, I prefer to have appman and am installed separately.

**Comment by ivan-hc** [comment]
You can keep follow the changes at https://github.com/ivan-hc/AM/pull/1036

**Comment by ochi12** [comment]
Yay!

**Comment by ivan-hc** [comment]
From a simple suggestion, I've converted AM in something new

@ochi12 if you want to test it in preview, run the following commands:
```
am --devmode-enable
am -s
```
and then use AM normally.

You can start with the options you have suggested, and then you can try all the others.

AppMan users should not notice differences, AM users instead will do anything.

To exit the developer mode, run:
```
am --devmode-disable
```

Your feedback will be precious to speedup the release.

I stick in "dev" as a pre-release waiting until everything will work perfectly.

Let me know.

**Comment by ivan-hc** [comment]
https://github.com/ivan-hc/AM/releases/tag/9

I think this discussion can be closed now. Thanks for this amazing idea!
